core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -521,11 +542,21 @@ class DefaultMessageFormatter extends MessageFormatter {
     if (props.containsKey("line.separator"))
       lineSeparator = props.getProperty("line.separator").getBytes(StandardCharsets.UTF_8)
     // Note that `toString` will be called on the instance returned by `Deserializer.deserialize`
-    if (props.containsKey("key.deserializer"))
+    if (props.containsKey("key.deserializer")) {
       keyDeserializer = Some(Class.forName(props.getProperty("key.deserializer")).newInstance().asInstanceOf[Deserializer[_]])
+      // Instantiate the inner key class if `default.windowed.key.serde.inner` is specified
```

Comment:
Would it be more general to just pass the whole `props` in the `configure` call so that if user used some other customized serde that needs configured with some properties, they can still do the same pattern as

```
--property value.deserializer=MyOwnDeserializer
--property my.own.deserializer.propOne=valueOne
--property my.own.deserializer.propTwo=valueTwo
```

Other properties that are not required will simply be ignored by the deserializer's configure function.